client/client.go

```diff
@@ -372,26 +362,26 @@ func Configure(logger hclog.Logger, providerConfig interface{}) (schema.ClientMe
 			&ec2.DescribeRegionsInput{AllRegions: aws.Bool(false)},
 			func(o *ec2.Options) {
 				o.Region = defaultRegion
-				if len(awsConfig.Regions) > 0 {
-					o.Region = awsConfig.Regions[0]
+				if len(localRegions) > 0 && !wildcardAllRegions {
+					o.Region = localRegions[0]
 				}
 			})
 		if err != nil {
 			return nil, fmt.Errorf("failed to find disabled regions for account %s. AWS Error: %w", accountID, err)
 		}
-		client.regions = filterDisabledRegions(client.regions, res.Regions)
+		account.Regions = filterDisabledRegions(localRegions, res.Regions, wildcardAllRegions)
```

Comment:
My personal preference is to remove the need for `wildcardAllRegions` as this can be inferred from `localRegions` within `filterDisabledRegions` - this way it will remove any potential bugs and/or inconsistencies of `localRegions` and `wildcardAllRegions` variables